src/bin/miri.rs

```diff
@@ -123,7 +122,47 @@ fn compile_time_sysroot() -> Option<String> {
     })
 }
 
+/// Execute a compiler with the given CLI arguments and callbacks.
+fn run_compiler(mut args: Vec<String>, callbacks: &mut (dyn rustc_driver::Callbacks + Send)) {
```

Comment:
Oh that's a nice convenience function. I wonder if we can upstream the non-sysroot part of it

Reply:
Yeah I wondered the same actually.
Any proposals for a name, given that `rustc_driver::run_compiler` is already taken?